### PR TITLE
ci: replace release notes generator action with one maintained by stepsecurity

### LINKED ISSUE #3267: Update release-notes-generator-action to step-security version
`.github/workflows/release-automation.yml` uses a non step-security maintaned version of action `release-notes-generator-action` it should get updated.

### PR DESCRIPTION
**Pull request by PavelSBorisov**
**Description**:
The release-automation.yml workflow uses a release notes generator action that has an equivalent maintained by StepSecurity.
This PR replaces it with the StepSecurity variant.

**Related issue(s)**:

Fixes #3267 

**Notes for reviewer**:
N/A
**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/workflows/release-automation.yml [hunk 1]
```diff
@@ -100,7 +100,7 @@ jobs:
 
       - name: Create Release Notes
         if: ${{ steps.milestone.outputs.milestone_id != '' }}
-        uses: Decathlon/release-notes-generator-action@98423db7024696a339f3988ac8a2b051c5860741 # v3.1.6
+        uses: step-security/release-notes-generator-action@d7cdbb310d4aab8d98f273f4ae20fdd7cb055799 # v3.1.6
```
Comment (andrewb1269hg) [inline .github/workflows/release-automation.yml:103]: Does the version number in the comment need to change as well?

**Review by andrewb1269hg — APPROVED** [review]
Looks good.

**Comment by andrewb1269hg** [comment]
@Nana-EC the codecov for this repo is below 80% which is causing failure of the codecov check. How would you like to resolve this issue?

@rbarkerSL for SA.

**Review by quiet-node — APPROVED** [review]

**Comment by Nana-EC** [comment]
> @Nana-EC the codecov for this repo is below 80% which is causing failure of the codecov check. How would you like to resolve this issue?
> 
> @rbarkerSL for SA.

In a non-blocking fashion.
We'll open a ticket to get the codecov back up. Thanks

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**